csrc/mmdeploy/net/ipu/ipu_net.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include "ipu_net.h"
+
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "mmdeploy/core/logger.h"
+#include "mmdeploy/core/model.h"
+#include "mmdeploy/core/utils/formatter.h"
+
+namespace mmdeploy::framework {
+
+mmdeploy::DataType IPUNet::ipu_type_convert(const popef::DataType& ipu_type) {
+  mmdeploy::DataType mtype;
+  if (ipu_type == popef::DataType::F32) {
+    mtype = mmdeploy::DataType::kFLOAT;
+  } else if (ipu_type == popef::DataType::F16) {
+    mtype = mmdeploy::DataType::kHALF;
+  } else if (ipu_type == popef::DataType::F8) {
+    mtype = mmdeploy::DataType::kINT8;
+  } else {
+    throw std::invalid_argument(
+        "invalid data type for IPU backend, current legit is : fp32, fp16, fp8");
+  }
+  return mtype;
+}
+
+void IPUNet::copy_output(const model_runtime::TensorMemory& from, Tensor& to) {
+  if (from.data_size_bytes != to.byte_size()) {
+    MMDEPLOY_ERROR("output tensor size not match from size {} to size {}", from.data_size_bytes,
+                   to.byte_size());
+    return;
+  }
+  int size = from.data_size_bytes;
+
+  MMDEPLOY_DEBUG("copy output total byte_size {}", size);
+
+  char* from_ptr = static_cast<char*>(from.data.get());
+
+  char* pto = to.data<char>();
+  for (int i = 0; i < size; i++) {
+    pto[i] = *(from_ptr + i);
+  }
+}
+
+void IPUNet::copy_input(const Tensor& from, model_runtime::TensorMemory& to) {
+  if (from.byte_size() != to.data_size_bytes) {
+    MMDEPLOY_ERROR("input tensor size not match  from size {} to size {} ", from.byte_size(),
+                   to.data_size_bytes);
+    return;
+  }
+  int size = from.byte_size();
+
+  MMDEPLOY_DEBUG("copy input total byte_size {}", size);
+
+  char* to_ptr = static_cast<char*>(to.data.get());
+
+  const char* pfrom = from.data<char>();
+  for (int i = 0; i < size; i++) {
+    *(to_ptr + i) = pfrom[i];
+  }
+}
+
+IPUNet::~IPUNet() {}
+
+Result<void> IPUNet::Init(const Value& args) {
+  auto& context = args["context"];
+  auto name = args["name"].get<std::string>();
+  auto model = context["model"].get<Model>();
+  OUTCOME_TRY(auto config, model.GetModelConfig(name));
+
+  batch_per_step = config.batches_per_step;
+
+  mconfig.device_wait_config =
+      model_runtime::DeviceWaitConfig(std::chrono::seconds{600}, std::chrono::seconds{1});
+  model_runner = std::make_unique<model_runtime::ModelRunner>(config.net, mconfig);
```

Comment:
The model dir looks like below.
```
.
├── deploy.json
├── detail.json
├── end2end
│   ├── end2end.onnx.optimized.onnx
│   └── executable.popef
├── end2end.onnx
└── pipeline.json
```

The deploy.json looks like this:
```
{
    "version": "0.12.0",
    "task": "Classifier",
    "models": [
        {
            "name": "resnet",
            "net": "end2end.popef",
            "weights": "",
            "backend": "ipu",
            "precision": "FP32",
            "batch_size": 1,
            "dynamic_shape": false
        }
    ],
    "customs": []
}
```
the value of `config.net` is end2end.popef, but the popef file is located in end2end/executable.popef which is not match. Moreover, the initialize of ModelRunner take absolute model path, but the config.net is relative path. You can refer the load of coreml model https://github.com/open-mmlab/mmdeploy/blob/master/csrc/mmdeploy/net/coreml/coreml_net.mm#L284-L286

Reply:
added abs path, to test the correctness